packages/boxel/addon/components/boxel/input/time/index.gts

```diff
@@ -0,0 +1,388 @@
+  }
+
+  get meridian() {
+    if (this.args.value) {
+      return this.args.value.getHours() < 12 ? 'am' : 'pm';
+    }
+
+    return undefined;
+  }
+
+  get hourOptions(): TimeOption[] {
+    return [...Array(12).keys()].map((value) => {
+      return {
+        value,
+        display: value === 0 ? '12' : value.toString()
+      };
+    });
+  }
+
+  get minuteOptions(): TimeOption[] {
+    return [...Array(60 / this.minuteInterval).keys()].map(n => n * this.minuteInterval ).map((value) => {
+      return {
+        value,
+        display: (value < 10 ? '0' : '') + value.toString()
+      };
+    });
+  }
+
+  get keyboardFocus(): KeyboardFocus {
+    let focusedEl = document.activeElement;
+    if (!focusedEl) {
+      return 'none';
+    }
+    if (!focusedEl.hasAttribute('data-keyboard-focus')) {
+      return 'none';
+    }
+    return focusedEl.getAttribute('data-keyboard-focus') as KeyboardFocus;
+  }
+
+  set keyboardFocus(val: KeyboardFocus) {
+    if (this.keyboardFocus !== val) {
+      let targetEl = this.containerElement?.querySelector(`[data-keyboard-focus="${val}"]`) as HTMLUListElement;
+      targetEl?.focus();
+    }
+  }
+
+  @action setHour(hourNum: number){
+    let { value } = this.args;
+    let effectiveDate = value ? new Date(value.getTime()) : new Date();
+    if (effectiveDate.getHours() > 11) {
+      hourNum = hourNum + 12;
+    }
+    let dateMs = effectiveDate.setHours(hourNum);
+    this.args.onChange?.(new Date(dateMs));
+  }
+
+  @action setMinute(minuteNum: number) {
+    minuteNum = Math.floor(minuteNum / this.minuteInterval) * this.minuteInterval; // round down to nearest interval
+    let { value } = this.args;
+    let effectiveDate = value ? new Date(value.getTime()) : new Date();
+    let dateMs = effectiveDate.setMinutes(minuteNum);
+    this.args.onChange?.(new Date(dateMs));
+  }
+
+  @action setMeridian(newMeridian: 'am'|'pm'){
+    if (this.meridian == newMeridian) {
+      return;
+    }
+    let { value } = this.args;
+    let effectiveDate = value ? new Date(value.getTime()) : new Date();
+    let hourNum;
+    if (newMeridian === 'am') {
+       hourNum = effectiveDate.getHours() - 12;
+    } else {
+      hourNum = effectiveDate.getHours() + 12;
+    }
+    let dateMs = effectiveDate.setHours(hourNum);
+    this.args.onChange?.(new Date(dateMs));
+  }
+
+  decrementHour() {
+    if (this.hour === undefined) {
+      this.setHour(11);
+      return;
+    }
+    if (this.hour > 0) {
+      this.setHour(this.hour - 1);
+      return;
+    }
+    if (this.hour === 0) {
+      this.setHour(11);
+      this.setMeridian(this.meridian == 'am' ? 'pm' : 'am');
+    }
+  }
+
+  incrementHour() {
+    if (this.hour === undefined) {
+      this.setHour(0);
+      return;
+    }
+    if (this.hour < 11) {
+      this.setHour(this.hour + 1);
+      return;
+    }
+    if (this.hour === 11) {
+      this.setHour(0);
+      this.setMeridian(this.meridian == 'am' ? 'pm' : 'am');
```

Comment:
nit: this could be it's own function since it is used twice, and maybe take leverage of the toggleMeridian function 🤔  Also shouldn't this check be 3 equals ?